File: app/src/main/java/com/martinlaizg/geofind/data/access/database/dao/UserDAO.java
package com.martinlaizg.geofind.data.access.database.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.martinlaizg.geofind.data.access.database.entities.User;

import java.util.List;

@Dao
public interface UserDAO {

	@Insert(onConflict = OnConflictStrategy.REPLACE)
	void insert(User user);

	@Update
	void update(User user);

	@Query("SELECT * FROM users WHERE id = :user_id")
	User getUser(int user_id);

}

File: app/src/main/java/com/martinlaizg/geofind/data/access/database/dao/UserDAO.kt
package com.martinlaizg.geofind.data.access.database.dao

import androidx.room.*
import com.martinlaizg.geofind.data.access.database.entities.User

@Dao
interface UserDAO {
	@Insert(onConflict = OnConflictStrategy.REPLACE)
	fun insert(user: User?)

	@Update
	fun update(user: User?)

	@Query("SELECT * FROM users WHERE id = :userId")
	fun getUser(userId: Int): User?
}